### ISSUE TITLE
Show private ideas count on my ideas

### ISSUE THREAD

**Issue by ezl** [ezl/dailyidea.com#848]
<img width="913" alt="image" src="https://user-images.githubusercontent.com/157627/103818900-fbe8fc80-502e-11eb-818d-0cd0ad9b756e.png">

The count at the top and the number of ideas is not matching.

I THINK this is because the count is showing the total number of ideas, but the display is only showing public ideas. (@vedmant, can you please confirm if what I said is true or false?).

If this is true, I think we should do one of these things:

(A) in the top summary count, show "2 ideas (1 private)"
(B) ONLY count public ideas (i.e. show "1 idea")

Personally, I lean towards (A). @Watson-B what do you think?

**Comment by ikosovec** [comment]
I can confirm that I have 2 ideas, of which 1 is private on my profile that's being referenced in this ticket.

**Comment by Watson-B** [comment]
@ezl @ikosovec I think both work, but I favor Option A since it highlights a paid feature. There are some issues with displaying the breakdown between public and private on the tab for smaller screen widths though. 

![image](https://user-images.githubusercontent.com/65913664/103821879-e7533700-5023-11eb-95cb-97155684975e.png)


Maybe just do the breakdown up top (the line right below the name) and then just have Ivan's Ideas on the tab?

![image](https://user-images.githubusercontent.com/65913664/103822056-41ec9300-5024-11eb-80a5-099c37f4b341.png)

Thoughts?

**Comment by ezl** [comment]
@Watson-B it's like my head is a tree and you're those little cookie elves. (yep, your design is perfect)

cc: @acerra86

**Comment by vedmant** [comment]
Yes, if you view other user it shows total number of ideas, including private, but in the list only public ideas are visible. 

It will however require backend edits to make it show only public ideas, currently there is just an increment field "ideasCreated" in the users table, so it doesn't count ideas, just increments this number. To make this work I'll need to do quite some changes on backend to make it count private and public ideas separately plus create some routine to update existing users counts.

**Comment by ezl** [comment]
@vedmant -- If I understand:

1. We currently have a key for total ideas, which is incremented whenever an idea is created, and decremented when an idea is deleted.
2. We are retrieving the number of ideas in the UI by just getting this number, not doing a query (select count(*) from ideas where user=username)

To implement showing public vs private, therefore, we would need to either.

(A) add ANOTHER counter that we access (so we have a public ideas counter and private ideas counter, which we then need to also add increment/decrement on make public vs make private action in addition to creation/deletion)
(B) start actually querying and counting to retrieve this info.

Is that correct?

If so, IMO the idea of adding another counter sounds like more dependent actions that cause potential multiple sources of truth and complexiity -- so I'd prefer option (B).

However, I also don't want to do much backend work if it involves significant change, so I'm likely to wait until we do a backend rewrite for this functionality.

Thoughts?

**Comment by vedmant** [comment]
It would makes sense for SQL database, but as I understand counting in
DynamoDb each time is quite expensive process. I'm not an expert however.

**Comment by ezl** [comment]
Understood. I don't want to invest in the appsync backend anymore for non-critical issues.

basically, my current decision logic on whether to implement a backend change right now is:


(A) is it critical / very important AND important that it happens SOON? If so, we can consider making the change to the AWS backend.

(B) If not, let's wait until we change to a different backend. This is something that's executed easily with django with a sql backend (`Idea.objects.filter(private=True).count()`).

So given what you're saying, let's just pause this until the future.

@Watson-B - thanks for creating the design. We'll be using this when we implement.
@ikosovec I'm moving this to icebox for execution at a future time after a backend rewrite.
